dockstore-integration-testing/src/test/java/io/dockstore/client/cli/GeneralWorkflowIT.java

```diff
@@ -131,6 +131,93 @@ private Workflow manualRegisterAndPublish(WorkflowsApi workflowsApi, String work
+        workflow = workflowsApi.refresh(workflow.getId(), false);
+        WorkflowVersion updatedVersion = workflow.getWorkflowVersions().stream().filter(workflowVersion -> Objects.equals(workflowVersion.getName(), versionOfInterest)).findFirst().get();
+        assertNotNull(updatedVersion.getCommitID());
+        assertNotEquals(versionOfInterest + " version should be updated (different dbupdatetime)", oldVersion.getDbUpdateDate(), updatedVersion.getDbUpdateDate());
+
+        // When the commit ID is different, a refresh should occur
+        oldVersion = workflow.getWorkflowVersions().stream().filter(workflowVersion -> Objects.equals(workflowVersion.getName(), versionOfInterest)).findFirst().get();
+        testingPostgres.runUpdateStatement("update workflowversion set commitid = 'dj90jd9jd230d3j9' where name = '" + versionOfInterest + "'");
+        workflow = workflowsApi.refresh(workflow.getId(), false);
+        updatedVersion = workflow.getWorkflowVersions().stream().filter(workflowVersion -> Objects.equals(workflowVersion.getName(), versionOfInterest)).findFirst().get();
+        assertNotNull(updatedVersion.getCommitID());
+        assertNotEquals(versionOfInterest + " version should be updated (different dbupdatetime)", oldVersion.getDbUpdateDate(), updatedVersion.getDbUpdateDate());
+
+        // Updating the workflow should make the version not synced, a refresh should refresh all versions
+        workflow.setWorkflowPath(incorrectDescriptorPath);
+        workflow = workflowsApi.updateWorkflow(workflow.getId(), workflow);
+        workflow.getWorkflowVersions().forEach(workflowVersion -> assertFalse(workflowVersion.isSynced()));
+        workflow = workflowsApi.refresh(workflow.getId(), false);
+
+        // All versions should be synced and updated
+        workflow.getWorkflowVersions().forEach(workflowVersion -> assertTrue(workflowVersion.isSynced()));
+        workflow.getWorkflowVersions().forEach(workflowVersion -> Objects.equals(workflowVersion.getWorkflowPath(), incorrectDescriptorPath));
+
+        // Update the version to have the correct path
+        WorkflowVersion testBothVersion = workflow.getWorkflowVersions().stream().filter(workflowVersion -> Objects.equals(workflowVersion.getName(), versionOfInterest)).findFirst().get();
+        testBothVersion.setWorkflowPath(correctDescriptorPath);
+        List<WorkflowVersion> versions = new ArrayList<>();
+        versions.add(testBothVersion);
+        workflowsApi.updateWorkflowVersion(workflow.getId(), versions);
+
+        // Refresh should only update the version that is not synced
+        workflow = workflowsApi.getWorkflow(workflow.getId(), "");
+        testBothVersion = workflow.getWorkflowVersions().stream().filter(workflowVersion -> Objects.equals(workflowVersion.getName(), versionOfInterest)).findFirst().get();
+        assertTrue("Version should not be synced", !testBothVersion.isSynced());
```

Comment:
Codacy found an issue: [assertTrue(!expr) can be replaced by assertFalse(expr)](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6348842)